### PR TITLE
Enhance getAuthorizationURL function 

### PR DESCRIPTION
**Pull request by movinsilva**
## Purpose
- The purpose of this PR is to enhance the getAuthorizationURL function by introducing support to build authorization post functions.

## Approach
- The logic for building parameters of the getAuthorizationURL function is abstracted out and exposed as a new function called **getAuthorizationURLParams**.

## Related Issue
- [Enhance getAuthorizationURL Function to Support POST Requests](https://github.com/asgardeo/asgardeo-auth-js-core/issues/250)

### CODE REVIEW

File: lib/src/client.ts [hunk 1]
```diff
@@ -104,6 +104,7 @@ export class AsgardeoAuthClient<T> {
         cryptoUtils: CryptoUtils,
         instanceID?: number
     ): Promise<void> {
+        console.log("Newest Local PACKAGE!!");
```
Comment (pavinduLakshan) [inline lib/src/client.ts:107]: Do we need this log here?

File: lib/src/client.ts [hunk 2]
```diff
@@ -169,6 +170,54 @@ export class AsgardeoAuthClient<T> {
         return AsgardeoAuthClient._instanceID;
     }
 
+    /**
+   * This is an async method that returns a Promise that resolves with the authorization URL parameters.
+   *
+   * @param config - (Optional) A config object to force initialization and pass
+   * custom path parameters such as the fidp parameter.
```
Comment (pavinduLakshan) [inline lib/src/client.ts:177]:
```suggestion
   * @param config - (Optional) A config object to force initialization and pass
   * custom path parameters such as the `fidp` parameter.
```

File: lib/src/client.ts [hunk 3]
```diff
@@ -169,6 +170,54 @@ export class AsgardeoAuthClient<T> {
...
+   * auth.getAuthorizationURLParams().then((params)=>{
+   *  // console.log(params);
+   * }).catch((error)=>{
+   *  // console.error(error);
+   * });
+   * ```
+   *
+   * {@link https://github.com/asgardeo/asgardeo-auth-js-sdk/tree/master#getAuthorizationURLParams}
+   *
+   * @preserve
```
Comment (pavinduLakshan) [inline lib/src/client.ts:194]: What is the usage of this doc tag here?

File: lib/src/client.ts [hunk 4]
```diff
@@ -169,6 +170,54 @@ export class AsgardeoAuthClient<T> {
...
+   */
+    public async getAuthorizationURLParams(
+        config?: GetAuthURLConfig,
+        userID?: string
+    ): Promise<Map<string, string>> {
+        const authRequestConfig: GetAuthURLConfig = { ...config };
+
+        delete authRequestConfig?.forceInit;
```
Comment (pavinduLakshan) [inline lib/src/client.ts:202]: If we are always deleting the forceInit param, why would we need to pass that in `config`?

File: lib/src/core/authentication-core.ts [hunk 5]
```diff
@@ -62,82 +62,102 @@ export class AuthenticationCore<T> {
...
+  
+        const authorizeRequestParams: Map<string, string> = new Map<
+        string,
+        string
+      >();
+  
         authorizeRequestParams.set("response_type", "code");
         authorizeRequestParams.set("client_id", configData.clientID);
-
+
```
Comment (pavinduLakshan) [inline lib/src/core/authentication-core.ts:78]: Remove the additional space characters. Please check and fix the other places as well.

```suggestion

```

File: lib/src/client.ts (hunk 1 above)
Comment (movinsilva) [reply lib/src/client.ts:107]: Ack and thanks. We don't need that log.

File: lib/src/core/authentication-core.ts (hunk 5 above)
Comment (movinsilva) [reply lib/src/core/authentication-core.ts:78]: ack.

File: lib/src/client.ts (hunk 3 above)
Comment (movinsilva) [reply lib/src/client.ts:193]: @link tag is providing a link to the documentation. Refer [docs.](https://jsdoc.app/tags-inline-link)

@preserve tag is a custom tag defined. Refer [tsdoc.json](https://github.com/asgardeo/asgardeo-auth-js-core/blob/6bb00d5a478f532da3261d24e625131ca6b7ccec/lib/tsdoc.json#L5)

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**